### ISSUE TITLE
Use the built-in `AggregateError` when possible

### ISSUE THREAD

**Issue by aleclarson** [radashi-org/radashi#75]
If `globalThis.AggregateError` exists, use that instead of our polyfill. Alternatively, remove `AggregateError` altogether and expect the end user to polyfill it themselves.

## Related issue

https://github.com/sodiray/radash/issues/392

**Comment by MarlonPassos-git** [comment]
I will work on it.

**Comment by MarlonPassos-git** [comment]
Hi @aleclarson 

This issue can be easily resolved with a small modification to the file:

```typescript
class AggregateErrorPolyfill extends Error {
  errors: Error[]
  constructor(errors: Error[] = []) {
    super()
    const name = errors.find(e => e.name)?.name ?? ''
    this.name = `AggregateError(${name}...)`
    this.message = `AggregateError with ${errors.length} errors`
    this.stack = errors.find(e => e.stack)?.stack ?? this.stack
    this.errors = errors
  }
}

export const AggregateError = globalThis?.["AggregateError"] ?? AggregateErrorPolyfill
```

However, this can be a bit more complicated due to the current `tsconfig` configuration, where `globalThis` might not have the `AggregateError` type definitions.

![image](https://github.com/user-attachments/assets/5a4d3a81-6d7c-4e53-8647-4aa5a9736f87)

We could resolve this in a couple of ways:

1. **Using a workaround:**

   ```typescript
   // @ts-ignore
   export const AggregateError: typeof AggregateErrorPolyfill = globalThis?.["AggregateError"] ?? AggregateErrorPolyfill
   ```

2. **Modifying the `tsconfig`:**

   Add `"ES2021.Promise"` to the `lib` configuration:

   ```json
   "lib": ["es2020", "ES2021.Promise"],
   ```

   Then define `AggregateError` with the appropriate type:

   ```typescript
   export const AggregateError: AggregateErrorConstructor = globalThis?.["AggregateError"] ?? AggregateErrorPolyfill
   ```

Regarding tests:

The current tests check the implementation of `AggregateErrorPolyfill`, which might differ from how the native `AggregateError` works.

![image](https://github.com/user-attachments/assets/3badb4ba-e09f-4a1b-8400-d2a85e303d50)
![image](https://github.com/user-attachments/assets/f979ccb3-3abb-4eaa-81a4-81ab32ffae31)
![image](https://github.com/user-attachments/assets/a685393a-8266-4415-9f86-8ed852078067)

With the current way we import the function in our tests, we will end up importing the native implementation.

However, it is possible to simulate an environment where `AggregateError` is not available and force the import of the polyfill version with:

```typescript
vi.stubGlobal('AggregateError', undefined)
const { AggregateError } = await import('radashi')
```

But this would cause another type error:

![image](https://github.com/user-attachments/assets/a7d2bb5e-76f5-4fbb-a5c5-45a156b98c32)

In this case, we would need to either use `@ts-ignore` or modify the `tsconfig` to accommodate the test files.

That being said, it would be useful to add two more tests to ensure that when `globalThis.AggregateError` is undefined, our polyfill is used, and when it exists, the value of our stub is correctly returned. 

What are your thoughts on this?

**Comment by aleclarson** [comment]
> Add `"ES2021.Promise"` to the `lib` configuration:

Let's put `/// <reference lib="es2021.promise" />` at the top of AggregateError module so we can keep tabs on why something beyond `es2020` is needed.



> ```ts
> export const AggregateError: AggregateErrorConstructor = globalThis?.["AggregateError"] ?? AggregateErrorPolyfill
> ```

I think it's fine to assume `globalThis` exists. The new `pnpm lint:compat` command doesn't complain.

Also, let's use `typeof globalThis.AggregateError` as the type and inline the class declaration like so:
```ts
export const AggregateError: typeof globalThis.AggregateError =
  globalThis.AggregateError ??
  class AggregateError {…}
```

> With the current way we import the function in our tests, we will end up importing the native implementation.
> 
> However, it is possible to simulate an environment where `AggregateError` is not available and force the import of the polyfill version with:
> 
> ```ts
> vi.stubGlobal('AggregateError', undefined)
> const { AggregateError } = await import('radashi')
> ```

I will take care of this part. Let me know when you open the PR 👍

**Comment by aleclarson** [comment]
Fixed in #116
